calculator_test.go

```diff
@@ -5,20 +5,82 @@ import (
 	"testing"
 )
 
-func TestAdd(t *testing.T) {
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+func TestCalculator(t *testing.T) {
+
+	var tests = []struct {
+		function func(float64, float64) float64
+		want     float64
+		input    []float64
+	}{
+		{function: calculator.Add, input: []float64{2, 2}, want: 4},
+		{function: calculator.Subtract, input: []float64{2, 4}, want: 2},
+		{function: calculator.Multiply, input: []float64{2, 4}, want: 8},
+	}
+	for _, tt := range tests {
+		got := tt.function(tt.input[0], tt.input[1])
+		if tt.want != got {
+			t.Errorf("want %f, got %f", tt.want, got)
+		}
+
+	}
+}
+
+func TestDivide(t *testing.T) {
+	var tests = []struct {
+		a             float64
+		b             float64
+		want          float64
+		errorExpected bool
+	}{
+		{
+			a:             4,
+			b:             0,
+			want:          999,
+			errorExpected: true},
+		{
+			a:             4,
+			b:             2,
+			want:          2,
+			errorExpected: false,
+		},
+	}
+	for _, tt := range tests {
+		got, err := calculator.Divide(tt.a, tt.b)
+		if tt.errorExpected && err == nil {
+			t.Fatal("want err/0, got nil")
+		}
+		if !tt.errorExpected && err != nil {
+			t.Fatal("Unexpected error")
```

Comment:
Again, let's be more explicit:

```suggestion
			t.Fatalf("Unexpected error: %v", err)
```

Reply:
(Right now the only possible error is the divide by zero error, but this test will still be around in ten years time when the codebase has been greatly expanded and extended, and it probably now calculates the orbits of satellites or something. So we want to make as few assumptions as possible in the test code.)